src/sched/basic.c

```diff
@@ -107,31 +106,25 @@ static void sched_run(ABT_sched sched)
     event_freq = p_data->event_freq;
     num_pools  = p_data->num_pools;
     pools      = p_data->pools;
+    ABT_unit unit = ABT_UNIT_NULL;
 
     while (1) {
-        CNT_INIT(run_cnt, 0);
-
-        /* Execute one work unit from the scheduler's pool */
-        for (i = 0; i < num_pools; i++) {
-            ABT_pool pool = pools[i];
-            ABTI_pool *p_pool = ABTI_pool_get_ptr(pool);
-            /* Pop one work unit */
-            ABT_unit unit = ABTI_pool_pop(p_pool);
-            if (unit != ABT_UNIT_NULL) {
+        for (int j = 0; j < num_pools; j++) {
+            ABTI_pool *p_pool = ABTI_pool_get_ptr(pools[j]);
+            ++pop_count;
+            if ((unit = ABTI_pool_pop(p_pool)) != ABT_UNIT_NULL) {
                 ABTI_xstream_run_unit(&p_local, p_xstream, unit, p_pool);
-                CNT_INC(run_cnt);
                 break;
             }
         }
-
-        if (++work_count >= event_freq) {
+        /* if we attempted event_freq pops, check for events */
+        if (pop_count >= event_freq) {
             ABTI_xstream_check_events(p_xstream, sched);
-            ABT_bool stop = ABTI_sched_has_to_stop(&p_local, p_sched,
-                                                   p_xstream);
-            if (stop == ABT_TRUE)
+            if (ABTI_sched_has_to_stop(&p_local, p_sched, p_xstream) 
```

Comment:
Remove the trailing whitespace.

Reply:
I don't think it has been removed. Could you please confirm it again?

Reply:
Could you please clarify which line has the trailing white space?

Reply:
```
if (ABTI_sched_has_to_stop(&p_local, p_sched, p_xstream)[ ]<=
```

Reply:
Removed.